### PR TITLE
Get logits as numpy array

### PR DESCRIPTION
**Pull request by KeremP**
I believe this could address https://github.com/nomic-ai/pygpt4all/issues/3.

logits are passed to a callback function at each iteration as a numpy array.
```c++
for (int i = embd.size(); i < embd_inp.size() + params.n_predict; i++) {
        // predict
        if (embd.size() > 0) {
            const int64_t t_start_us = ggml_time_us();

            if (!gptj_eval(model, params.n_threads, n_past, embd, logits, mem_per_token)) {
                printf("Failed to predict\n");
                return 1;
            }
            // collect logits for each token
            py::array_t<float> _logits = py::array_t<float>{model.hparams.n_vocab, logits.data(), py::none()};
            logits_callback(_logits);
            t_predict_us += ggml_time_us() - t_start_us;
        }
...
}
```
```python
def _call_logits_callback(self, logits: np.ndarray):
        """
        Internal logits_callback that saves the logit representation at each token.
        :return: None
        """
        self.logits.append(logits.tolist())
        
        if Model._logits_callback is not None:
            Model._logits_callback(logits)
```
Vectors are appended to  ```self.logits```  and can be dumped to .npy by calling ```model.braindump("output_path") ```

### CODE REVIEW

**Comment by absadiki** [comment]
That's great @KeremP.
Thank you very much.
